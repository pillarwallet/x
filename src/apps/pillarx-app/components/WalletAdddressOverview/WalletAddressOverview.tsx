import { useEffect, useState } from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';

// images
import { MdCheck } from 'react-icons/md';

// components
import CopyIcon from '../../images/copy-icon.svg';
import RandomAvatar from '../RandomAvatar/RandomAvatar';
import Body from '../Typography/Body';

type WalletAddressOverviewProps = {
  address: string;
};

const WalletAddressOverview = ({ address }: WalletAddressOverviewProps) => {
  const [isCopied, setIsCopied] = useState(false);

  useEffect(() => {
    if (isCopied) {
      const timer = setTimeout(() => {
        setIsCopied(false);
      }, 3000);

      return () => clearTimeout(timer);
    }

    return undefined;
  }, [isCopied]);

  return (
    <div className="desktop:flex tablet:flex mobile:hidden mb-[54px] gap-2.5 items-center">
      <div className="bg-medium_grey rounded-full w-10 h-10 overflow-hidden">
        <RandomAvatar name={address} />
      </div>
      <Body>
        {address.substring(0, 6)}...{address.substring(address.length - 5)}
      </Body>
      <CopyToClipboard text={address} onCopy={() => setIsCopied(true)}>
        <div className="w-4 h-4">
          {isCopied ? (
            <MdCheck style={{ width: 'full', height: 'full' }} />
          ) : (
            <img
              src={CopyIcon}
              alt="copy-adress-icon"
              className="cursor-pointer"
              style={{ width: 'full', height: 'full' }}
            />
          )}
        </div>
      </CopyToClipboard>
    </div>
  );
};

export default WalletAddressOverview;
