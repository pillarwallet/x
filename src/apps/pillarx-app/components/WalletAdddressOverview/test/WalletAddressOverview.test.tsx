import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import WalletAddressOverview from '../WalletAddressOverview';

describe('<WalletAddressOverview />', () => {
  const address = '0x1234567890';
  const mockSetIsCopied = vi.fn();
  const mockUseState = vi.spyOn(React, 'useState');

  beforeEach(() => {
    mockUseState.mockReturnValue([false, mockSetIsCopied]);
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('renders correctly and matches snapshot', () => {
    const tree = renderer
      .create(<WalletAddressOverview address={address} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct icon and address', () => {
    const tree = renderer
      .create(<WalletAddressOverview address={address} />)
      .toJSON() as ReactTestRendererJSON;

    const profileIcon = tree.children?.find(
      (child) =>
        typeof child === 'object' &&
        child.props.className === 'bg-medium_grey p-2.5 rounded-full w-10 h-10'
    ) as ReactTestRendererJSON;

    const addressText = tree.children?.find(
      (child) =>
        typeof child === 'object' &&
        child.props.children ===
          `${address.substring(0, 6)}...${address.substring(address.length - 5)}`
    ) as ReactTestRendererJSON;

    expect(profileIcon).not.toBeNull();
    expect(addressText).not.toBeNull();
  });
});
