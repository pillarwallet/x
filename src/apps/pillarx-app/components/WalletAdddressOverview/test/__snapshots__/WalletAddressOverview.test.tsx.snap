// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletAddressOverview /> renders correctly and matches snapshot 1`] = `
<div
  className="desktop:flex tablet:flex mobile:hidden mb-[54px] gap-2.5 items-center"
>
  <div
    className="bg-medium_grey rounded-full w-10 h-10 overflow-hidden"
  >
    <svg
      className="rounded-md"
      data-testid="random-avatar"
      fill="none"
      role="img"
      viewBox="0 0 80 80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <mask
        height={80}
        id=":r0:"
        maskUnits="userSpaceOnUse"
        width={80}
        x={0}
        y={0}
      >
        <rect
          fill="#FFFFFF"
          height={80}
          width={80}
        />
      </mask>
      <g
        mask="url(#:r0:)"
      >
        <rect
          fill="#C271B4"
          height={80}
          width={80}
        />
        <path
          d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
          fill="#C20D90"
          filter="url(#filter_:r0:)"
          transform="translate(-2 2) rotate(-26 40 40) scale(1.5)"
        />
        <path
          d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
          fill="#92A1C6"
          filter="url(#filter_:r0:)"
          style={
            {
              "mixBlendMode": "overlay",
            }
          }
          transform="translate(7 7) rotate(39 40 40) scale(1.5)"
        />
      </g>
      <defs>
        <filter
          colorInterpolationFilters="sRGB"
          filterUnits="userSpaceOnUse"
          id="filter_:r0:"
        >
          <feFlood
            floodOpacity={0}
            result="BackgroundImageFix"
          />
          <feBlend
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          />
          <feGaussianBlur
            result="effect1_foregroundBlur"
            stdDeviation={7}
          />
        </filter>
      </defs>
    </svg>
  </div>
  <p
    className="text-base font-medium undefined"
  >
    0x1234
    ...
    67890
  </p>
  <div
    className="w-4 h-4"
    onClick={[Function]}
  >
    <img
      alt="copy-adress-icon"
      className="cursor-pointer"
      src={
        {
          "ReactComponent": {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "default": "copy-icon.svg",
        }
      }
      style={
        {
          "height": "full",
          "width": "full",
        }
      }
    />
  </div>
</div>
`;
